src/ui.cc

```diff
@@ -417,6 +419,35 @@ int ReadFlagsClean(int* argc, char*** argv, Execution::Options* options) {
   return ReadTargets(argc, argv, options);
 }
 
+int ReadFlagsCommands(int* argc, char*** argv, Execution::Options* options) {
+  // Step back argv to include 'clean' so that getopt will
+  // work correctly since it starts reading at position 1.
+  ++(*argc);
+  --(*argv);
```

Comment:
Yikes...

Reply:
Sorry, I'm probably just missing something obvious since I don't normally use getopt - what's bad about this?

Reply:
Manipulating argc and argv is just gross in general, that's all.

Not saying that your code is wrong.